nixos/modules/config/networking.nix

```diff
@@ -161,22 +156,30 @@ in
       '';
     }];
 
+    # These entries are required for "hostname -f" and to resolve both the
+    # hostname and FQDN correctly:
     networking.hosts = {
-      "127.0.0.1" = [ "localhost" ];
-    } // optionalAttrs (cfg.hostName != "") {
-      "127.0.1.1" = [ cfg.hostName ];
+      "127.0.0.1" = localhostHostname;
```

Comment:
This is `127.0.0.2` inside `nss-myhostname`.

Reply:
Should I also use `127.0.0.2` then? The reason seems to be the following:
```
src/nss-myhostname/nss-myhostname.c:/* We use 127.0.0.2 as IPv4 address. This has the advantage over
src/nss-myhostname/nss-myhostname.c- * 127.0.0.1 that it can be translated back to the local hostname. For
src/nss-myhostname/nss-myhostname.c- * IPv6 we use ::1 which unfortunately will not translate back to the
src/nss-myhostname/nss-myhostname.c- * hostname but instead something like "localhost" or so. */
```

E.g. for `/etc/hosts` using `127.0.1.1` is more typical and our current default so maybe we should use that instead.